src/state/stake/hooks-sushi.ts

```diff
@@ -163,3 +163,42 @@ export function useInfiniteScroll(items: any[]): [number, Dispatch<number>] {
   useEffect(() => setItemsDisplayed(10), [items.length])
   return [itemsDisplayed, setItemsDisplayed]
 }
+
+export function getMultipleContracts(
+  addressList: string[],
+  ABI: any,
+  library: Web3Provider,
+  account?: string
+): Contract[] {
+  const invalidAddress = addressList.find(address => !isAddress(address) || address === AddressZero)
+  if (invalidAddress) {
+    throw Error(`Invalid 'address' parameter '${invalidAddress}'.`)
+  }
+  const contracts = addressList.map(address => new Contract(address, ABI, getProviderOrSigner(library, account) as any))
+  return contracts
```

Comment:
WET code, let's make it DRY:

```suggestion
  return addressList.map(address => getContract(address, ABI, library, account))
```

Reply:
true, ty!